lib/ex_admin/ex_admin.ex

```diff
@@ -180,6 +180,20 @@ defmodule ExAdmin do
     |> Keyword.get(resource_model)
   end
 
+
+  def get_registered_by_association(resource, assoc_name) do
+    resource_model = resource.__struct__
+    assoc_model = case resource_model.__schema__(:association, assoc_name) do
+      %{through: [link1, link2]} ->
+        resource |> Ecto.build_assoc(link1) |> Ecto.build_assoc(link2) |> Map.get(:__struct__)
+      %{queryable: assoc_model} ->
+        assoc_model
+      nil ->
```

Comment:
I see your point about nil. That indicates the association does  not exist. I was worried about not matching nay of the three clauses, and throwing a complicated exception. But perhaps your three match conditions will catch all possible associations. The probably all have a 'queryable' field.